docs/property-graph-model.adoc

```diff
@@ -0,0 +1,75 @@
+:numbered:
+
+= Property Graph Model
+ifdef::env-github,env-browser[:outfilesuffix: .adoc]
+
+Cypher is a graph query language, but it doesn't operate on any arbitrary definition of graph; it operates on a Property Graph.
+A Property Graph may be defined in graph theoretical terms as a directed, vertex-labeled, edge-labeled multigraph with self-edges, where edges have their own identity.
+See Wikipedia's definitions for reference:
+
+- link:https://en.wikipedia.org/wiki/Multigraph#Directed_multigraph_.28edges_with_own_identity.29[Directed multigraph].
+- link:https://en.wikipedia.org/wiki/Multigraph#Labeling[Labeled multigraph].
+
+In a Property Graph, the following elements may exist:
+
+* Entity
+** Node
+** Relationship
+* Path
+* Token
+** Label
+** Relationship type
+** Property key
+* Property
+
+=== Entity
+
+- An entity has a unique, comparable identity, which defines whether two entities are equal or not.
+- An entity is assigned a set of properties, which are each uniquely identified in the set by their respective property keys.
+
+==== Node
+
+- A _node_ is the basic entity of the graph, with the unique attribute of being able to exist in and of itself.
+- A node is assigned a set of unique labels.
```

Comment:
I thought that was clear as a set may be empty. Do you think this needs additional mention? Would the following be better (which we discussed):

> A node may be assigned a set of unique labels.